flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperator.java

```diff
@@ -405,6 +411,8 @@ public void merge(W mergeResult,
 		// windowAssigner is event time and current timestamp + allowed lateness no less than element timestamp
 		if (isSkippedElement && lateDataOutputTag != null && isElementLate(element)) {
```

Comment:
the if-else conditions are duplicated and inefficient, and can be further combined as

```java
if(isSkippedElement && isElementLate(element)) {
    if(lateDataOutputTag != null) {
        sideOutput(element);
    }
    this.lostDataCount.inc();
}
```

Reply:
it looks better this way, i will adjust here  :)